make buildtest fails on OSX

```
examples/hello-world > make buildtest
find: -printf: unknown primary or operator
/bin/sh: -c: line 0: syntax error near unexpected token `esac'
/bin/sh: -c: line 0: `if [ -z "${JENKINS_URL}" ] && tput colors 2>&1 > /dev/null; then  GREEN='\033[1;32m'; RED='\033[1;31m'; PURPLE='\033[1;35m'; RESET='\033[0m';     ECHO='/bin/echo -e'; else       GREEN=''; RED=''; PURPLE=''; RESET=''; ECHO='/bin/echo'; fi; BUILDTESTOK=true; rm -rf "$BINDIRBASE"; for BOARD in ; do         RIOTNOLINK=$(case '  ' in *" ${BOARD} "*) echo 1; esac);        ${ECHO} -n "Building for ${BOARD} ";    [ -n "${RIOTNOLINK}" ] && ${ECHO} -n "(no linking) ";  for NTH_TRY in 1 2; do          ${ECHO} -n ".. ";               LOG=$(env -i HOME=${HOME} PATH=${PATH} BOARD=${BOARD} RIOTBASE=${RIOTBASE} RIOTBOARD=${RIOTBOARD} RIOTCPU=${RIOTCPU} BINDIRBASE=${BINDIRBASE} RIOTNOLINK=${RIOTNOLINK} RIOT_VERSION=${RIOT_VERSION} make -j3 2>&1 >/dev/null) ;                if [ "${?}" = "0" ]; then                       ${ECHO} "${GREEN}success${RESET}";              elif [ -n "${RIOT_DO_RETRY}" ] && ${BUILDTESTOK} && [ ${NTH_TRY} = 1 ]; then                   ${ECHO} -n "${PURPLE}retrying${RESET} ";                        continue;               else                    ${ECHO} "${RED}failed${RESET}";                         echo "${LOG}" | grep -v -E '^make(\[[[:digit:]]])?:';                  BUILDTESTOK=false;              fi;             break;  done; done; ${BUILDTESTOK}'
make: *** [buildtest] Error 2
```

--- a/Makefile.buildtests
+++ b/Makefile.buildtests
@@ -31,6 +31,7 @@ endif
         info-buildsizes-diff info-build info-boards-supported \
         info-features-missing info-boards-features-missing
 
+OS := $(shell uname)
 
 COLOR_GREEN  :=
 COLOR_RED    :=
@@ -43,7 +44,12 @@ ifeq (, ${JENKINS_URL})
     COLOR_RED    := \033[1;31m
     COLOR_PURPLE := \033[1;35m
     COLOR_RESET  := \033[0m
-    COLOR_ECHO   := /bin/echo -e
+    ifeq ($(OS),Darwin)
+      COLOR_ECHO   := echo -e
+      SHELL=bash
+    else
+      COLOR_ECHO   := /bin/echo -e
+    endif
   endif
 endif
 
@@ -52,7 +58,7 @@ buildtest:
 	BUILDTESTOK=true; \
 	rm -rf "$$BINDIRBASE"; \
 	for BOARD in $$($(MAKE) -s info-boards-supported); do \
-		RIOTNOLINK=$$(case ' $(BOARD_INSUFFICIENT_RAM) ' in *" $${BOARD} "*) echo 1; esac); \
+		RIOTNOLINK=$$(echo $(BOARD_INSUFFICIENT_RAM) | grep $${BOARD} 2>&1 >/dev/null && echo 1); \
 		${COLOR_ECHO} -n "Building for $${BOARD} "; \
 		[ -n "$${RIOTNOLINK}" ] && ${COLOR_ECHO} -n "(no linking) "; \
 		for NTH_TRY in 1 2; do \
@@ -229,7 +235,7 @@ ifneq (, $(filter info-boards-supported info-boards-features-missing info-build,
     endif
   endef
 
-  BOARDS ?= $(shell find $(RIOTBOARD)/* -maxdepth 0 -type d \! -name *-common -printf '%f ')
+  BOARDS ?= $(shell find $(RIOTBOARD)/* -maxdepth 0 -type d \! -name "*-common" -exec basename {} \;)
   BOARDS := $(filter $(if $(BOARD_WHITELIST), $(BOARD_WHITELIST), %), $(BOARDS))
   BOARDS := $(filter-out $(BOARD_BLACKLIST), $(BOARDS))
 

